src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -271,34 +458,212 @@ contract StrategyManagerUnitTests is Test, Utils {
         strategyManager.depositIntoStrategy(IStrategy(address(reenterer)), dummyToken, amount);
     }
 
-    function testDepositIntoStrategyWithSignatureSuccessfully(uint256 amount, uint256 expiry) public {
-        // min shares must be minted on strategy
-        cheats.assume(amount >= 1);
-
-        address staker = cheats.addr(privateKey);
-        // not expecting a revert, so input an empty string
-        string memory expectedRevertMessage;
-        _depositIntoStrategyWithSignature(staker, amount, expiry, expectedRevertMessage);
-    }
-
-    function testDepositIntoStrategyWithSignatureReplay(uint256 amount, uint256 expiry) public {
-        // min shares must be minted on strategy
-        cheats.assume(amount >= 1);
-        cheats.assume(expiry > block.timestamp);
+    function test_Revert_WhenTokenSafeTransferFromReverts() external {
+        // replace 'dummyStrat' with one that uses a reverting token
+        dummyToken = IERC20(address(new Reverter()));
+        dummyStrat = _deployNewStrategy(dummyToken, strategyManager, pauserRegistry, dummyAdmin);
 
-        address staker = cheats.addr(privateKey);
-        // not expecting a revert, so input an empty string
+        // whitelist the strategy for deposit
+        cheats.startPrank(strategyManager.owner());
```

Comment:
nit: can just use cheats.prank(address) if there's one call to be pranked. Won't need cheats.stopPrank() if so

Reply:
Added changes here but left in certain areas. See my commit message https://github.com/Layr-Labs/eigenlayer-contracts/commit/d7c322b5a0bbd86e10c45377f3c89c0633c0d89e